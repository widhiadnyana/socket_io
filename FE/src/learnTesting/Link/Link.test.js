// Link.react.test.js
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { fireEvent, render, screen } from "@testing-library/react";
import Link from "./Link";

test("Link changes the class when hovered", () => {
    const { asFragment, getByTestId } = render(
        <Link page='http://www.facebook.com'>Facebook</Link>
    );

    let tree = asFragment(<Link page='http://www.facebook.com'>Facebook</Link>);
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    fireEvent.mouseEnter(getByTestId("link"));
    // re-rendering
    tree = asFragment(<Link page='http://www.facebook.com'>Facebook</Link>);
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    fireEvent.mouseLeave(getByTestId("link"));
    // re-rendering
    tree = asFragment(<Link page='http://www.facebook.com'>Facebook</Link>);
    expect(tree).toMatchSnapshot();
});
