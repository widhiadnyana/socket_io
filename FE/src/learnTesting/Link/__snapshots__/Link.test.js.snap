// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<DocumentFragment>
  <a
    class="normal"
    data-testid="link"
    href="http://www.facebook.com"
  >
    Facebook
  </a>
</DocumentFragment>
`;

exports[`Link changes the class when hovered 2`] = `
<DocumentFragment>
  <a
    class="hovered"
    data-testid="link"
    href="http://www.facebook.com"
  >
    Facebook
  </a>
</DocumentFragment>
`;

exports[`Link changes the class when hovered 3`] = `
<DocumentFragment>
  <a
    class="normal"
    data-testid="link"
    href="http://www.facebook.com"
  >
    Facebook
  </a>
</DocumentFragment>
`;
